src/IntegratesQueryBuilder.js

```diff
@@ -0,0 +1,50 @@
+import { Mixin } from 'mixwith'
+
+/**
+ * @needs MakesRequests
+ */
+export default Mixin((superclass) => class extends superclass
+{
+    beforeRequest(request) {
+        if (superclass.hasOwnProperty('beforeRequest')) {
+            superclass.beforeRequest(request)
+        }
+
+        if (!request.builder) {
+            return request
+        }
+
+        if (typeof request.builder !== 'function') {
+            throw new Error('Builder property must be a function.')
+        }
+
+        let queryBuilderModule
+
+        try {
+            queryBuilderModule = require('js-query-builder')
```

Comment:
Good, that seems to be the way to define optional dependencies. We just need to set it up in the `externals` webpack options and according to [this issue](https://github.com/webpack/webpack/issues/339) it should recognise this as an optional external.

Reply:
I'm not sure if it is necessary to define it as external dependency somewhere.

Reply:
`require()` works like a charm.

Reply:
Okay cool, so you don't have any errors when running `npm run build`?

Reply:
I haven't tried `npm run build` yet.
But I have no errors with `npm run test` both with and without `js-query-builder` being installed.